utils/s2n_mem.c

```diff
@@ -88,15 +87,16 @@ int s2n_realloc(struct s2n_blob *b, uint32_t size)
         GUARD(s2n_free(b));
         S2N_ERROR(S2N_ERR_MLOCK);
     }
-    b->mlocked = 1;
 
     return 0;
 }
 
+#include<stdio.h>
+
 int s2n_free(struct s2n_blob *b)
 {
     int munlock_rc = 0;
-    if (b->mlocked) {
+    if (use_mlock && b->data) {
```

Comment:
This looks good, but we're now assuming that every call to s2n_free had a previous call to s2n_alloc. Is it worth writing the scaffolding code to enforce this? I'm leaning towards no. Alloc/free is a basic enough primitive for us to assume developers will use the pattern.